text/0148-sampling-probability.md

```diff
@@ -0,0 +1,744 @@
+both for producers (i.e., SDKs) and for processors (i.e., Collectors),
+without losing the ability to (at least coarsely) monitor the whole
+system.
+
+Sampling builds on results from probability theory, most significantly
+the concept of expected value.  Estimates drawn from probability
+samples are *random variables* that, when correct procedures are
+followed, accurately reflect their true value, making them unbiased.
+Unbiased samples can be used for after-the-fact analysis.  We can
+answer questions such as "what fraction of events had property X?"
+using the fraction of events in the sample that have property X.
+
+This document outlines how producers and consumers of sample telemetry
+data can convey estimates about the total count of telemetry events,
+without conveying information about how the sample was computed, using
+a quantity known as **adjusted count**.  In common language, a
+"one-in-N" sampling scheme emits events with adjusted count equal to
+N.  Adjusted count is the expected value of the number of events in
+the population represented by an individual sample event.
+
+## Examples
+
+In all of these examples, the use of probability sampling leads to an
+attribute named like `sampling.sampler_name.adjusted_count`.
+Consumers of spans, metrics, and logs annotated with adjusted counts
+are able to calculate accurate statistics about the population of
+events, at a basic level, without knowing details about the sampling
+configuration.
+
+### Span sampling
+
+<details> 
+<summary> Example use-cases for probability sampling of spans
+generally involve generating metrics from spans.</summary>
+
+#### Sample spans to Counter Metric
+
+For every span it receives, the example processor will synthesize
+metric data as though a Counter instrument named `S.count` for span
+named `S` had been incremented once per span at the original `Start()`
+call site.
+
+This processor will add the adjusted count of each span to the
+instrument (e.g., `Add(adjusted_count, labels...)`)  for every span it
+receives, logically taking place at the start or end time of the span.
+
+#### Sample spans to Histogram Metric
+
+For every span it receives, the example processor will synthesize
+metric data as though a Histogram instrument named `S.duration` for
+span named `S` had been observed once per span at the original `End()`
+call site.
+
+The OpenTelemetry Metric data model does not support histogram buckets
+with non-integer counts, which forces the use of integer adjusted
+counts here (i.e., 1-in-N sampling rates where N is an integer).
+
+Logically spaking, this processor will observe the span's duration its
+adjusted count number of times for every span it receives, at the end
+time of the span.
+
+#### Sample span rate limiting
+
+A collector processor will introduce a slight delay in order to ensure
+it has received a complete frame of data, during which time it
+maintains a fixed-size buffer of input spans.  If the number of spans
```

Comment:
Will this buffer only contain finished spans, or it can have spans that are started but not finished?

Reply:
I added "finished" to describe these spans.